File: factors.c
#include <stdio.h>

void factorize_number(int n) {
    for (int i = 2; i <= n / 2; ++i) {
        if (n % i == 0) {
            printf("%d=%d*%d\n", n, i, n / i);
            return;  // Assuming you only want one factorization per line
        }
    }
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
        fprintf(stderr, "Usage: %s <file>\n", argv[0]);
        return 1;
    }

    FILE *file = fopen(argv[1], "r");
    if (file == NULL) {
        perror("Error opening file");
        return 1;
    }

    int number;
    while (fscanf(file, "%d", &number) == 1) {
        factorize_number(number);
    }

    fclose(file);

    return 0;
}

File: factors.py
import sys

def factorize_number(n):
    # Find the factors of n
    factors = []
    for i in range(2, n//2 + 1):
        if n % i == 0:
            factors.append((i, n//i))
    return factors

def main():
    if len(sys.argv) != 2:
        print("Usage: {} <file>".format(sys.argv[0]))
        sys.exit(1)

    input_file = sys.argv[1]

    with open(input_file, 'r') as file:
        for line in file:
            number = int(line.strip())
            factorizations = factorize_number(number)
            for factorization in factorizations:
                print("{}={}*{}".format(number, factorization[0], factorization[1]))

if __name__ == "__main__":
    main()
